### ISSUE TITLE
bug: case sensitive check for the headers in the `Access-Control-Request-Headers`

### ISSUE THREAD

**Issue by mikalaiyurkin** [pgjones/quart-cors#1]

**Comment by mikalaiyurkin** [comment]
The code here expecting that the `allow_headers` and `request_headers` are the iterables with strings in the same registry

```
response.access_control_allow_headers = HeaderSet(
    set(allow_headers).intersection(set(request_headers))
)
```

Instead there should be used smth like this

```
response.access_control_allow_headers = HeaderSet(
    allow_headers.as_set().intersection(request_headers.as_set())
)
```

**Comment by pgjones** [comment]
Fixed in 458afde4632a3de22bf66bb6287d7c274c032c9a
